invenio_rdm_records/resources/resources.py

```diff
@@ -89,16 +93,30 @@ def p(route):
     @request_extra_args
     @request_read_args
     @request_view_args
-    @response_handler()
+    @custom_response_handler()
     def read(self):
         """Read an item."""
-        item = self.service.read(
-            g.identity,
-            resource_requestctx.view_args["pid_value"],
-            expand=resource_requestctx.args.get("expand", False),
-            # allows to access deleted record if permissions match
-            include_deleted=resource_requestctx.args.get("include_deleted", False),
-        )
+        try:
+            item = self.service.read(
+                g.identity,
+                resource_requestctx.view_args["pid_value"],
+                expand=resource_requestctx.args.get("expand", False),
+                # allows to access deleted record if permissions match
+                include_deleted=resource_requestctx.args.get("include_deleted", False),
+            )
+        except NoResultFound:
+            # If the parent pid is being used we can get the id of the latest record and redirect
+            latest_version = self.service.read_latest(
```

Comment:
Not sure how I feel about this being handled here, instead of e.g. inside the service... But we've kind of "normalized" that `NoResultFound` (which is a pretty low-level exception) is a valid exception that the resource error handlers take care of...

I would prefer either that the service catches the `NoResultFound` in an appropriate place, and then re-raises a new "high-level" error that error handlers use to redirect as needed (cc @zzacharo I think you did something similar for redirecting from parent to latest version).

Reply:
We are using the same behaviour at the moment, see here: https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/views/decorators.py#L175

As agreed I cretead the following issue to address this: https://github.com/inveniosoftware/invenio-app-rdm/issues/2539